components/common/hooks/useGetUsageDeltas.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useMemo} from 'react';
+
+import {CloudUsage} from '@mattermost/types/cloud';
+
+import useGetUsage from './useGetUsage';
+import useGetLimits from './useGetLimits';
+
+export default function useGetUsageDeltas(): CloudUsage {
+    const usage = useGetUsage();
+    const cloudLimits = useGetLimits();
+    const limits = cloudLimits[0];
+
+    const usageDelta = useMemo(() => (
+        {
+            files: {
+                totalStorage: usage.files.totalStorage - (limits.files?.total_storage || 0),
+                totalStorageLoaded: usage.files.totalStorageLoaded,
+            },
+            messages: {
+                history: usage.messages.history - (limits.messages?.history || 0),
+                historyLoaded: usage.messages.historyLoaded,
+            },
+            boards: {
+                cards: usage.boards.cards - (limits.boards?.cards || 0),
+                cardsLoaded: usage.boards.cardsLoaded,
+            },
+            teams: {
+                active: usage.teams.active - (limits.teams?.active || 0),
+                teamsLoaded: usage.teams.teamsLoaded,
+            },
```

Comment:
If the team was archived already because the trial ended, this will always show as not exceeding the team. We may need a special check to get if there are any teams archived due to limits (aka your new column in the db)

Reply:
I plan to address this in a future PR. As it stands, this isn't even hooked up to the backend in any way, all hard coded.

Reply:
Do you have a ticket to add the team usage part to this? Is the idea to somehow capture what teams are archived because of limits?

Reply:
The backend code for teams usage lives here: https://github.com/mattermost/mattermost-server/pull/20281

Hooking that up to the front-end is happening here: https://github.com/mattermost/mattermost-webapp/pull/10387 as  its necessary to make the front-end for https://mattermost.atlassian.net/browse/MM-44475 work